meson.build

```diff
@@ -0,0 +1,1629 @@
+project('mpv',
+        'c',
+        license: ['GPL2+', 'LGPL2.1+'],
+        version: '0.33',
+        default_options: [
+            'buildtype=debugoptimized',
+            'c_std=c11',
+            'warning_level=1',
+        ]
+)
+
+cc = meson.get_compiler('c')
+
+
+# essential dependencies
+ffmpeg = {
+    'name': 'ffmpeg',
+    'deps': [dependency('libavutil', version: '>= 56.12.100'),
+             dependency('libavcodec', version: '>= 58.12.100'),
+             dependency('libavformat', version: '>= 58.9.100'),
+             dependency('libswscale', version: '>= 5.0.101'),
+             dependency('libavfilter', version: '>= 7.14.100'),
+             dependency('libswresample', version: '>= 3.0.100')],
+}
+
+libass = dependency('libass', version: '>= 0.12.2')
+
+pthreads = dependency('threads')
+
+zlib = cc.find_library('z')
+
+dependencies = [ffmpeg['deps'],
+                libass,
+                pthreads,
+                zlib]
+
+iconv = find_program('iconv', required: get_option('iconv'))
+if not iconv.found() and not get_option('iconv').disabled()
```

Comment:
If the option is enabled, the find_program will immediately fail and not process the next line. If the option is disabled, this error() gets skipped. This error() only applies in the auto case, but the point of a feature option is that auto is "do a configure probe, and act like enabled if it is found, or disabled if it is not found".

Moreover, the build will already key off of iconv.found() to define the HAVE_ICONV macro, which is the single source of truth for building with iconv support. So this error message serves no purpose other than to turn "auto" into "required plus a custom error message", and the HAVE_ICONV macro is in actuality being selected based on whether or not the exact option `-Diconv=disabled` is passed.

It defeats the purpose of using a feature option / having an auto option.

Dumb question: why does the waf build check for iconv_open available as a symbol, (potentially using find_library('iconv') for systems where it's not in libc), but here you're checking for the command line program?

Also: if you would like `dependency('iconv')` we can add it to meson's built-in custom dependency lookups module, just like zlib and, as it happens, "intl" (which even depends on iconv).

Reply:
https://github.com/mesonbuild/meson/pull/9205

You may now do `dependency('iconv')` in (unreleased) meson 0.60